### PR TITLE
Fix completions error with forbidden body + adjust tests to fix

### LINKED ISSUE #44: [BUG] Chat example is failing with HTTP 422
**Describe the bug**
Example app is crashing with exception:
```
Unhandled exception:
MistralAIClientException: Request POST https://api.mistral.ai/v1/chat/completions failed with status code: 422, body: {"object":"error","message":{"detail":[{"type":"extra_forbidden","loc":["body","messages",0,"user","name"],"msg":"Extra inputs are not permitted","input":null},{"type":"extra_forbidden","loc":["body","messages",0,"user","tool_calls"],"msg":"Extra inputs are not permitted","input":null}]},"type":"invalid_request_error","param":null,"code":null}
#0      _requestJson (package:mistralai_client_dart/src/network/request.dart:22:7)
<asynchronous suspension>
#1      main (file:///Users/harizo/Downloads/staging/mistralai_client_dart/example/mistralai_client_dart_example.dart:21:26)
<asynchronous suspension>
```

**To Reproduce**
Run `dart examples/mistralai_client_dart_example.dart` in the root of package repository.

**Expected behavior**
Exception shouldn't happen and all requests should work as expected.

**Additional context**
Found during review in https://github.com/mistralai/platform-docs-public/pull/51

### PR DESCRIPTION
**Pull request by MarcinGruchala**
### Description

Fix completions error with forbidden body + adjust tests to fix
### Issue

Closes #44 

### Changes Made

- `includeIfNull`  to false where needed 
- adjust unit tests 

### Checklist

- [x] I have read the [CONTRIBUTING.md](https://github.com/nomtek/mistralai_client_dart/blob/main/CONTRIBUTING.md) document.
- [x] I have tested my changes.
- [x] I have added/updated documentation accordingly.

### CODE REVIEW

File: lib/src/models/chat_completion.dart [hunk 1]
```diff
@@ -73,8 +73,9 @@ class ChatMessage {
 
   final String role;
   final String content;
+  @JsonKey(includeIfNull: false)
```
Comment (gawi151) [inline lib/src/models/chat_completion.dart]: Wouldn't be better to add it to `@JsonSerializable(includeIfNull: false)` instead of adding keys? Wdyt?

File: test/src/mistralai_client_dart_base_chat_test.dart [hunk 2]
```diff
@@ -247,9 +246,7 @@ const chatCompletionParamsWithAllFieldsBody = '''
   "messages": [
     {
       "role": "user",
-      "content": "What is the best French cheese?",
-      "name": null,
-      "tool_calls": null
+      "content": "What is the best French cheese?"
```
Comment (gawi151) [inline test/src/mistralai_client_dart_base_chat_test.dart]: Ok so the name of the const is now not reflecting the actual body.

File: test/src/mistralai_client_dart_base_chat_stream_test.dart [hunk 3]
```diff
@@ -82,7 +82,6 @@ void main() {
               'role': 'role1',
               'content': 'content1',
               'name': 'name1',
-              'tool_calls': null,
```
Comment (gawi151) [inline test/src/mistralai_client_dart_base_chat_stream_test.dart:85]: I think we also need to add test just for checking if there is any null in the json. If there is then we have a bug.

**Review by gawi151 — CHANGES_REQUESTED** [review]
Few bits and pieces to improve.

File: lib/src/models/chat_completion.dart (hunk 1 above)
Comment (MarcinGruchala) [reply lib/src/models/chat_completion.dart]: Resolved in [6bc844c](https://github.com/nomtek/mistralai_client_dart/pull/45/commits/6bc844c69af070c078aa6aaa23ea36e417859095)

File: test/src/mistralai_client_dart_base_chat_test.dart (hunk 2 above)
Comment (MarcinGruchala) [reply test/src/mistralai_client_dart_base_chat_test.dart]: Resolved in [e824818](https://github.com/nomtek/mistralai_client_dart/pull/45/commits/e824818379c0768111aefed6a10c237da468dd83)

**Review by gawi151 — APPROVED** [review]
LGTM
